src/collections/Menu/Menu.js

```diff
@@ -81,10 +81,11 @@ export default class Menu extends Component {
 
     return Children.map(children, (child, i) => {
       const isItem = child.type === MenuItem
+      const isLink = _.has(child, 'props.href') || _.has(child, 'props.link') || _.has(child, 'props.onClick')
 
       if (isItem) {
         const onClick = (e) => {
-          this.handleItemClick(e, i)
+          if (isLink) this.handleItemClick(e, i)
           if (child.props.onClick) child.props.onClick(e, i)
         }
```

Comment:
![image](https://cloud.githubusercontent.com/assets/14183168/17836767/93b7c076-67a6-11e6-845c-74294814b63a.png)
![image](https://cloud.githubusercontent.com/assets/14183168/17836770/9cc73ad4-67a6-11e6-9654-3b37a3c424fd.png)

It seems that ESLint is more intelligent that you think :smiley_cat: